pkg/adsc/delta.go

```diff
@@ -385,6 +428,74 @@ func initWatch(typeURL string, resourceName string) Option {
 	}
 }
 
+func deltaMcpConfigHandlerForGvk(rev string, store model.ConfigStore, gvk config.GroupVersionKind) func(HandlerContext, string, string, *mcp.Resource, Event) {
+	return func(ctx HandlerContext, resourceName string, resourceVersion string, resourceEntity *mcp.Resource, event Event) {
+		// the format of mcp resource's name is "namespace/name"
+		// for delete case, the resource entity may be empty, the resource name is required
+		// for add/update case, extract the resource name and resource version from the resource entity
+		var ns, name string
+		var cfg *config.Config
+		if event == EventDelete {
+			nsn := strings.Split(resourceName, "/")
+			if len(nsn) != 2 {
+				ctx.Reject(fmt.Errorf("invalid resource name %s", resourceName))
+				return
+			}
+			ns, name, cfg = nsn[0], nsn[1], nil
+		} else {
+			newCfg, matchRev, err := mcpToPilot(resourceEntity, rev, false)
```

Comment:
This is a little complex. 

How about rewriting `mcpToPilot` return cfg if it is in revision, otherwise return nil

Confused by `strict`

Reply:
For delta, if the cfg is not in revision, it needs to be removed based on it's namespace and name.